BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj

```diff
@@ -23,9 +23,21 @@
     <PackageReference Include="Microsoft.Diagnostics.Tracing.EventRegister" Version="1.1.28" Condition="$(OS) == 'Windows_NT'">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0" />
   </ItemGroup>
 
+  <Choose>
+    <When Condition="'$(Redfield)' == 'True'">
+      <ItemGroup>
```

Comment:
Is it possible to use something simpler like:

```
<ItemGroup Condition="'$(Redfield)' == 'True'">
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.7.0" />
</ItemGroup>
```

or even:

```
<PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0" Condition="'$(Redfield)' != 'True'"/>
```

Reply:
or perhaps extract the version number, and set the version based on conditions in the common file (e.g. where `$(RedField)` is being defined:

```
<PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(SystemDiagnosticsDiagnosticSourcePkgVer)" />
```

Reply:
How about this one?

```
  <PropertyGroup Label="Package version for System.Diagnostics.DiagnosticSource">
    <SystemDiagnosticsDiagnosticSourcePkgVer>5.0.0</SystemDiagnosticsDiagnosticSourcePkgVer>
    <SystemDiagnosticsDiagnosticSourcePkgVer Condition="'$(Redfield)' == 'True'">4.7.0</SystemDiagnosticsDiagnosticSourcePkgVer>
  </PropertyGroup>

  <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(SystemDiagnosticsDiagnosticSourcePkgVer)" />
```

Reply:
I liked the explicitness of the `Choose` statement. To me, this communicates that these settings are coupled and not trivial to change.

But I don't have a strong opinion. Raj's suggestion is definitely cleaner!